### PR TITLE
[FLINK-22704][tests] Harden ZooKeeperHaServicesTest.testCleanupJobData

### PR DESCRIPTION
**Pull request by tillrohrmann**
This commit hardens the ZooKeeperHaServicesTest.testCleanupJobData by making sure that all
zNodes have been created before checking for their existence. This is done by waiting for
the leader information for the ResourceManager and the JobMaster.

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java [hunk 1]
```diff
@@ -224,23 +224,26 @@ private void runCleanupTestWithJob(
...
                     new LeaderRetrievalUtils.LeaderConnectionInfoListener();
-            resourceManagerLeaderRetriever.start(listener);
+            resourceManagerLeaderRetriever.start(resourceManagerLeaderListener);
             resourceManagerLeaderElectionService.start(
                     new TestingContender("foobar", resourceManagerLeaderElectionService));
             LeaderElectionService jobManagerLeaderElectionService =
                     zooKeeperHaServices.getJobManagerLeaderElectionService(jobId);
             jobManagerLeaderElectionService.start(
-                    new TestingContender("", jobManagerLeaderElectionService));
+                    new TestingContender("barfoo", jobManagerLeaderElectionService));
```
Comment (XComp) [inline flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java:235]:
```suggestion
                    new TestingContender("unused-jobmanager-address", jobManagerLeaderElectionService));
```
nit: I understand that you just followed the scheme of the other `TestingContender`. But I usually find it better to make the purpose of a certain String parameters visible if possible. That improves the readability of the test code. Without an IDE, I wouldn't be able to see right away what this parameter is used for if I'm not familiar with the `TestingContender` class.

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java [hunk 2]
```diff
@@ -224,23 +224,26 @@ private void runCleanupTestWithJob(
...
             LeaderRetrievalService jobManagerLeaderRetriever =
                     zooKeeperHaServices.getJobManagerLeaderRetriever(jobId);
-            jobManagerLeaderRetriever.start(
-                    new LeaderRetrievalUtils.LeaderConnectionInfoListener());
+            final LeaderRetrievalUtils.LeaderConnectionInfoListener jobMasterLeaderListener =
```
Comment (XComp) [inline flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java:238]: nit: `jobManagerLeaderListener` might be a better name for consistency purposes. But I understand that `jobMasterLeaderListener` is the more correct name.

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java [hunk 3]
```diff
@@ -224,23 +224,26 @@ private void runCleanupTestWithJob(
...
                     zooKeeperHaServices.getRunningJobsRegistry();
 
-            final LeaderRetrievalUtils.LeaderConnectionInfoListener listener =
+            final LeaderRetrievalUtils.LeaderConnectionInfoListener resourceManagerLeaderListener =
                     new LeaderRetrievalUtils.LeaderConnectionInfoListener();
-            resourceManagerLeaderRetriever.start(listener);
+            resourceManagerLeaderRetriever.start(resourceManagerLeaderListener);
             resourceManagerLeaderElectionService.start(
                     new TestingContender("foobar", resourceManagerLeaderElectionService));
             LeaderElectionService jobManagerLeaderElectionService =
```
Comment (XComp) [inline flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java:232]: nit: `final` is missing

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java [hunk 4]
```diff
@@ -224,23 +224,26 @@ private void runCleanupTestWithJob(
...
                     zooKeeperHaServices.getJobManagerLeaderElectionService(jobId);
             jobManagerLeaderElectionService.start(
-                    new TestingContender("", jobManagerLeaderElectionService));
+                    new TestingContender("barfoo", jobManagerLeaderElectionService));
             LeaderRetrievalService jobManagerLeaderRetriever =
```
Comment (XComp) [inline flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java:236]: nit: `final` is missing

**Review by XComp — APPROVED** [review]
The change looks good to me and the reasoning makes sense. I had some minor remarks to improve the readability of the test code.

IMHO, the test code could be also improved by grouping the lines more consistently:
```
            // create some Zk services to trigger the generation of paths
            final RunningJobsRegistry runningJobsRegistry =
                    zooKeeperHaServices.getRunningJobsRegistry();

            final LeaderRetrievalService resourceManagerLeaderRetriever =
                    zooKeeperHaServices.getResourceManagerLeaderRetriever();
            final LeaderElectionService resourceManagerLeaderElectionService =
                    zooKeeperHaServices.getResourceManagerLeaderElectionService();

            final LeaderRetrievalUtils.LeaderConnectionInfoListener resourceManagerLeaderListener =
                    new LeaderRetrievalUtils.LeaderConnectionInfoListener();
            resourceManagerLeaderRetriever.start(resourceManagerLeaderListener);
            resourceManagerLeaderElectionService.start(
                    new TestingContender("foobar", resourceManagerLeaderElectionService));

            LeaderRetrievalService jobManagerLeaderRetriever =
                    zooKeeperHaServices.getJobManagerLeaderRetriever(jobId);
            LeaderElectionService jobManagerLeaderElectionService =
                    zooKeeperHaServices.getJobManagerLeaderElectionService(jobId);

            final LeaderRetrievalUtils.LeaderConnectionInfoListener jobMasterLeaderListener =
                    new LeaderRetrievalUtils.LeaderConnectionInfoListener();
            jobManagerLeaderRetriever.start(jobMasterLeaderListener);
            jobManagerLeaderElectionService.start(
                    new TestingContender("barfoo", jobManagerLeaderElectionService));
```

All my comments are more of code-style-related issues. Feel free to address them or not. Thanks for coming up with a fix that fast.

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java:235]: Sounds good. I will change it.

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java (hunk 2 above)
Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java:238]: Ok, I think this is valid. Will change it.

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java (hunk 3 above)
Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java:232]: jup, will add it.

**Pull request closed without merging**